src/main/java/io/gatling/mojo/RecorderMojo.java

```diff
@@ -89,6 +86,17 @@ public class RecorderMojo extends AbstractGatlingMojo {
 
   @Override
   public void execute() throws MojoExecutionException, MojoFailureException {
+    if (simulationsFolder == null) {
+      // testCompileSourceRoots is expected to contain exactly one element (can be set by
+      // build.testSourceDirectory in the POM); but otherwise it is ambiguous and we fall back to
+      // the default Java test sources root.
+      if (mavenProject.getTestCompileSourceRoots().size() == 1) {
```

Comment:
IMHO, we should just always pick the first one instead.
In this case, we don't need the second branch either.